packages/embark-ipfs/src/upload.js

```diff
@@ -34,7 +34,11 @@ class IPFS {
         console.debug(cmd);
         shelljs.exec(cmd, {silent:true}, function(code, stdout, stderr){ // {silent:true}: don't echo cmd output so it can be controlled via logLevel
           console.log(stdout.green);
-          callback(stderr, stdout);
+          if (code) {
+            // stderr can sometimes be an error or the total uploaded size
+            return callback(stderr || __('IPFS exited with code %s', code.toString()));
+          }
+          callback(null, stdout);
```

Comment:
👍 much better to check status code here.

Just a side note regarding the comment:
> // stderr can sometimes be an error or the total uploaded size

Turns out that usually anything that's not actual program output is sent to stderr (warnings etc). So stderr can potentially include anything that shouldn't be processed by other programs.